pyflakes/test/test_imports.py

```diff
@@ -1093,6 +1093,26 @@ def test_concatenationAssignment(self):
         __all__ = ['a'] + ['b'] + ['c']
         ''', m.UndefinedExport, m.UndefinedExport, m.UndefinedExport, m.UnusedImport)
 
+    def test_all_with_attributes(self):
+        self.flakes('''
+        from foo import bar
+        __all__ = [bar.__name__]
+        ''')
+
+    def test_all_with_attributes_added(self):
+        self.flakes('''
+        from foo import bar
+        from bar import baz
+        __all__ = [bar.__name__] + [baz.__name__]
+        ''')
+
+    def test_all_mixed_attributes_and_strings(self):
+        self.flakes('''
+        from foo import bar
+        from foo import baz
+        __all__ = ['bar', baz.__name__]
+        ''')
```

Comment:
I've also (in the past) written (and continue to see others do)

```py
from foo import bar
from foo import baz

__all__ = [bar, baz]
```

Should we test for that too? I'm not sure we should necessarily do the right thing but I do wonder if we want to test the behaviour at least.

Reply:
I'll add a test that makes sure it doesn't crash.  That said, afaict it's not valid:

```python
import os

__all__ = [os]
```

```pycon
>>> from t import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: attribute name must be string, not 'module'
```